spec/requests/check_link_spec.rb

```diff
@@ -0,0 +1,115 @@
+require "rails_helper"
+
+RSpec.describe "check link path", type: :request do
+  let(:link) { "http://www.example.com" }
+
+  def check_link_path(query_params = {})
+    path = "/check-link"
+    path + (query_params.empty? ? "?#{query_params.to_query}" : "")
```

Comment:
I think `"/check-link" + ...` would look fine here, no need for the extra variable.